code/game/turfs/simulated/floor/plating.dm

```diff
@@ -269,6 +269,12 @@
 					var/mob/living/live = L.buckled
 					if("lava" in live.weather_immunities)
 						continue
+			if(ishuman(L))
```

Comment:
just give thrown things protection from lava, thrown OBJECTS already are, but lava still burns thrown mobs, fix that and you avoid snowflake.